stravalib/unithelper.py

```diff
@@ -2,28 +2,102 @@
 Unit Helper
 ==============
 Helpers for converting Strava's units to something more practical.
-
-These are really just thin wrappers to the brilliant 'units' python library.
 """
-from units import unit
-import units.predefined
+from numbers import Number
+from typing import Protocol, Union, runtime_checkable, Any
+
+import pint
+
+from stravalib.exc import warn_units_deprecated
+
+
+@runtime_checkable
+class UnitsQuantity(Protocol):
+    """
+    A type that represents the (deprecated) `units` Quantity. The `unit`
+    attribute in the units library consists of other classes, so this
+    representation may not be 100% backward compatible!
+    """
+    num: float
+    unit: str
+
+
+class Quantity:
+    """
+    Wraps `pint.Quantity`. Subclassing didn't work because pint has some heavily
+    customized attribute lookup
+    """
+
+    def __init__(self, q: pint.Quantity):
+        self.q = q
+
+    @property
+    def num(self):
+        warn_units_deprecated()
+        return self.q.magnitude
+
+    @property
+    def unit(self):
+        warn_units_deprecated()
+        return str(self.q.units)
+
+    def __int__(self):
+        return int(self.q.magnitude)
+
+    def __float__(self):
+        return float(self.q.magnitude)
+
+    def __getattr__(self, item):
+        return getattr(self.q, item)
+
+    def __str__(self):
+        return str(self.q)
+
+    def __repr__(self):
+        return repr(self.q)
+
+
+class UnitConverter:
```

Comment:
This class ensures we have a callable that behaves as the legacy `units`-based converters. It wraps the converted `pint.Quantity` object such that it can be interacted with as if it was a `units.Quantity` object.